### PR TITLE
[deckhouse] generate bashible context inside bashible-apiserver

### PR DESCRIPTION
**Pull request by yalosev**
## Description
Move context generation into bashible-apiserver

## Why we need it and what problem does it solve?
Bashible context Is too big and can't be handled with kubernetes secret (due to size limitations).
We want to pass only input values to bashible-server and generate a context in memory

## Changelog entries

```changes
module: deckhouse
type: fix
description: Move context generation into bashible-apiserver
note: bashible-apiserver will restart
```

### CODE REVIEW

File: modules/040-node-manager/images/bashible-apiserver/pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -150,3 +163,70 @@ func newBashibleInformerFactory(resync time.Duration, namespace, labelSelector s
...
+const (
+	configurationsSecretName      = "configuration-checksums"
+	configurationsSecretNamespace = "d8-cloud-instance-manager"
+)
+
+type checksumSecretUpdater struct {
+	clientset *kubernetes.Clientset
+}
+
+func (cs checksumSecretUpdater) OnCheckumUpdate(ngmap map[string][]byte) {
```
Comment (nabokihms) [inline modules/040-node-manager/images/bashible-apiserver/pkg/apiserver/apiserver.go:189]: Typo
```suggestion
func (cs checksumSecretUpdater) OnChecksumUpdate(ngmap map[string][]byte) {
```

File: modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go [hunk 2]
```diff
@@ -60,45 +108,240 @@ func (c *BashibleContext) subscribe(factory informers.SharedInformerFactory) cha
...
+}
+
+func (c *BashibleContext) runFilesParser() {
+	c.parseImageTagsFile()
+	c.parseVersionMapFile()
+
+	go c.runFilesWatcher()
+}
+
+func (c *BashibleContext) parseImageTagsFile() {
```
Comment (nabokihms) [inline modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go:134]: nit: file is called `images_tags`.

**Review by nabokihms — CHANGES_REQUESTED** [review]
Overall, it looks nice. 

Managed to find some NITs, though. Would you please check the comments when you have time?

File: modules/020-deckhouse/templates/registry.yaml [hunk 3]
```diff
@@ -9,7 +9,7 @@ metadata:
     helm.sh/resource-policy: keep
 {{- /* Helm checks changes only in .dockerconfigjson field, so if other field changes, helm don't change secret. Checksum resolve this. */}}
     checksum: {{ printf $.Values.global.modulesImages.registryDockercfg $.Values.global.modulesImages.registryAddress $.Values.global.modulesImages.registryPath $.Values.global.modulesImages.registryScheme $.Values.global.modulesImages.registryCA | sha256sum }}
-{{ include "helm_lib_module_labels" (list .) | indent 2 }}
+{{ include "helm_lib_module_labels" (list . (dict "app" "registry")) | indent 2 }}
```
[the remark this answers is not in the document]
Comment (yalosev) [reply modules/020-deckhouse/templates/registry.yaml:12]: added

File: modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go (hunk 2 above)
Comment (yalosev) [reply modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go]: +

File: modules/020-deckhouse/templates/rbac-for-us.yaml [hunk 4]
```diff
@@ -7,3 +7,32 @@ metadata:
   annotations:
     helm.sh/resource-policy: keep
 {{ include "helm_lib_module_labels" (list .) | indent 2 }}
+
+# RBAC for bashible - access to registry secret
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: bashible-apiserver-readregsecret
```
[the remark this answers is not in the document]
Comment (yalosev) [reply modules/020-deckhouse/templates/rbac-for-us.yaml]: It doesn't work:
```
    Linter error:
    
        🐒 [#MANIFEST053]
        	Message	- Role with prefix "bashible-apiserver" should be deployed in namespace "d8-cloud-instance-manager"
        	Object	- kind = Role ; name = bashible-apiserver-readregsecret ; namespace = d8-system
        	Value	- d8-system
        
        🐒 [#MANIFEST053]
        	Message	- Role in "templates/bashible-apiserver/rbac-for-us.yaml" should start with "bashible-apiserver" or "node-manager:bashible-apiserver"
        	Object	- kind = Role ; name = d8:bashible-apiserver-readconfig ; namespace = d8-cloud-instance-manager
        	Value	- d8:bashible-apiserver-readconfig
```

[the remark this answers is not in the document]
Comment (nabokihms) [reply modules/020-deckhouse/templates/rbac-for-us.yaml]: You can find the example here https://github.com/deckhouse/deckhouse/blob/41a6fd88f5877f1c66abf8c250bb40b6ff6d541a/modules/040-node-manager/templates/bashible/rbac-for-us.yaml#L43

**Review by nabokihms — APPROVED** [review]
Approved, everything seems ok. @konstantin-axenov, do you want to take a look, or can we merge this?

**Pull request merged**